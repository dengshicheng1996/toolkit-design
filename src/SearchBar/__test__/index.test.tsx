import * as React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { SearchBar } from '..'

describe('SearchBar', () => {
  it('SearchBar snapshot', () => {
    const component = shallow(<SearchBar placeholder="Search" maxLength={8} />)

    expect(toJSON(component)).toMatchSnapshot()
  })

  it('SearchBar showCancelButton snapshot', () => {
    const component = shallow(
      <SearchBar
        placeholder="Show Cancel Button"
        showCancelButton
        maxLength={24}
        data-y="111"
      />,
    )

    expect(toJSON(component)).toMatchSnapshot()
  })
})
