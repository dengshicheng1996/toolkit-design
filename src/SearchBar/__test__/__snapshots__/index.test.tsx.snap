// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar SearchBar showCancelButton snapshot 1`] = `
<form
  action="#"
  className="amd-search"
  onSubmit={[Function]}
>
  <div
    className="amd-search-input"
    style={Object {}}
  >
    <div
      className="amd-search-synthetic-ph"
    >
      <span
        className="amd-search-synthetic-ph-container"
      >
        <SearchOutline
          size="xs"
        />
        <span
          className="amd-search-synthetic-ph-placeholder"
          style={
            Object {
              "visibility": "visible",
            }
          }
        >
          Show Cancel Button
        </span>
      </span>
    </div>
    <input
      autoFocus={false}
      className="amd-search-value"
      data-y="111"
      disabled={false}
      maxLength={24}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Show Cancel Button"
      type="search"
      value=""
    />
    <Touchable
      onPress={[Function]}
    >
      <a
        className="amd-search-clear"
      >
        <CloseCircleFill
          size="xs"
        />
      </a>
    </Touchable>
  </div>
  <Touchable
    onPress={[Function]}
  >
    <div
      className="amd-search-cancel amd-search-cancel-show amd-search-cancel-anim"
    >
      取消
    </div>
  </Touchable>
</form>
`;

exports[`SearchBar SearchBar snapshot 1`] = `
<form
  action="#"
  className="amd-search"
  onSubmit={[Function]}
>
  <div
    className="amd-search-input"
    style={Object {}}
  >
    <div
      className="amd-search-synthetic-ph"
    >
      <span
        className="amd-search-synthetic-ph-container"
      >
        <SearchOutline
          size="xs"
        />
        <span
          className="amd-search-synthetic-ph-placeholder"
          style={
            Object {
              "visibility": "visible",
            }
          }
        >
          Search
        </span>
      </span>
    </div>
    <input
      autoFocus={false}
      className="amd-search-value"
      disabled={false}
      maxLength={8}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Search"
      type="search"
      value=""
    />
    <Touchable
      onPress={[Function]}
    >
      <a
        className="amd-search-clear"
      >
        <CloseCircleFill
          size="xs"
        />
      </a>
    </Touchable>
  </div>
  <Touchable
    onPress={[Function]}
  >
    <div
      className="amd-search-cancel amd-search-cancel-anim"
    >
      取消
    </div>
  </Touchable>
</form>
`;
